WordPress/gutenberg
# Visual and experience improvements to existing sub navigation flow

## Description

Implementation of https://github.com/WordPress/gutenberg/issues/22087

*Before*
![2020-05-06 14-25-56 2020-05-06 14_26_21](https://user-images.githubusercontent.com/205419/81176571-9a572680-8fa5-11ea-97fa-ec91a8baf75e.gif)

*After*
![2020-05-06 15-25-28 2020-05-06 15_25_43](https://user-images.githubusercontent.com/205419/81182509-2705e280-8fae-11ea-9884-b74c11d3d88e.gif)

## How has this been tested?
1. Open post editor
1. Add a navigation block with two levels of nested menus
1. Confirm the nested menus are only visible after clicking on them
1. Confirm everything looks good visually
1. Save the post
1. Open the post on the actual site (outside of the editor)
1. Confirm the menu works the same as before this PR and looks good

## Types of changes
Non-breaking changes

## Checklist:
